### PR TITLE
vcpkg for macOs 

### PR DESCRIPTION
**Pull request by daschuer**
This builds an environment for MacOS 
Traget specific overlays have been moved to overlays/x64-osc and overlays/x64-win

The qt version of MacOs is 5.12.4, Windows is on 5.15.2

### CODE REVIEW

File: overlay/ports/qt5-base/CONTROL [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Source: qt5-base
+Version: 5.12.4-6
```
Comment (Be-ing) [inline overlay/ports/qt5-base/CONTROL]: We are using Qt 5.12.3 in Mixxx 2.3. Hopefully there are no regressions with 5.12.4, but we'll have to test...

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -8,15 +8,19 @@ jobs:
       fail-fast: false
       matrix:
         include:
-          #- os: windows-2019
-          #  vcpkg_path: C:\mixxx-vcpkg
-          #  vcpkg_bootstrap: .\bootstrap-vcpkg.bat
-          #  vcpkg_triplet: x64-windows
-          # check_disk_space: Get-PSDrive
+          - os: windows-2019
```
Comment (Be-ing) [inline .github/workflows/build.yml]: You have removed libmad and libid3tag for Windows.

File: ports/portmidi/portfile.cmake [hunk 3]
```diff
@@ -54,13 +54,11 @@ if(VCPKG_TARGET_IS_WINDOWS)
     endif()
 else()
     if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
-        file(INSTALL ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/Release/portmidi_s.lib DESTINATION ${CURRENT_PACKAGES_DIR}/lib)
-        file(INSTALL ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/Debug/portmidi_s.lib DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib)
```
Comment (Be-ing) [inline ports/portmidi/portfile.cmake]: Does this break PortMidi on Windows?

File: overlay/ports/qt5-base/portfile.cmake [hunk 4]
```diff
@@ -163,14 +163,14 @@ elseif(VCPKG_TARGET_IS_OSX)
         string(REGEX MATCH "^[0-9][0-9]\.[0-9][0-9]*" VCPKG_OSX_DEPLOYMENT_TARGET ${VCPKG_OSX_DEPLOYMENT_TARGET})
         message(STATUS "Major.Minor OSX SDK Version: ${VCPKG_OSX_DEPLOYMENT_TARGET}")
         set(ENV{QMAKE_MACOSX_DEPLOYMENT_TARGET} ${VCPKG_OSX_DEPLOYMENT_TARGET})
-        if(${VCPKG_OSX_DEPLOYMENT_TARGET} GREATER "10.14") # Max Version supported by QT
-            message(STATUS "Qt 5.12.4 only support OSX_DEPLOYMENT_TARGET up to 10.14")
-            set(VCPKG_OSX_DEPLOYMENT_TARGET "10.14")
+        if(${VCPKG_OSX_DEPLOYMENT_TARGET} GREATER "10.15") # Max Version supported by QT. This version is defined in mkspecs/common/macx.conf as QT_MAC_SDK_VERSION_MAX
```
Comment (Be-ing) [inline overlay/ports/qt5-base/portfile.cmake]: Why is this change needed if we use 10.12 as the deployment target?

File: .github/workflows/build.yml (hunk 2 above)
Comment (daschuer) [reply .github/workflows/build.yml]: This commit is outdated.
Please compare to the final result.

File: triplets/x64-osx.cmake [hunk 5]
```diff
@@ -5,3 +5,7 @@ set(VCPKG_LIBRARY_LINKAGE static)
 set(VCPKG_CMAKE_SYSTEM_NAME Darwin)
 set(VCPKG_OSX_ARCHITECTURES x86_64)
 
+set(VCPKG_OSX_DEPLOYMENT_TARGET 10.12)
+set(VCPKG_C_FLAGS -mmacosx-version-min=10.12)
+set(VCPKG_CXX_FLAGS -mmacosx-version-min=10.12)
```
Comment (Be-ing) [inline triplets/x64-osx.cmake:10]: I would prefer to keep this in the GH Actions workflow file than hack vcpkg's source code. Alternatively we may make our own custom triplet.

File: .github/workflows/build.yml (hunk 2 above)
Comment (Be-ing) [reply .github/workflows/build.yml]: This PR is difficult to review because of the intermediate changes that got reverted later. Can you squash it down to a few commits, perhaps:
1. portmidi changes
2. macOS 10.12 target changes
3. GH Actions workflow changes

File: overlay/ports/qt5-base/portfile.cmake (hunk 4 above)
Comment (daschuer) [reply overlay/ports/qt5-base/portfile.cmake]: This is bugfix from upstream.

File: triplets/x64-osx.cmake (hunk 5 above)
Comment (daschuer) [reply triplets/x64-osx.cmake:10]: Yes, Ok.

File: overlay/ports/qt5-base/CONTROL (hunk 1 above)
Comment (Be-ing) [reply overlay/ports/qt5-base/CONTROL]: Please bump the version to 5.12.4-7 to distinguish it from the upstream vcpkg version.

File: overlay/ports/qt5-base/portfile.cmake (hunk 4 above)
Comment (Be-ing) [reply overlay/ports/qt5-base/portfile.cmake]: I'm unclear where it comes from. Is it backported from a later Qt version upstream?

**Comment by Be-ing** [comment]
Somehow uploading the logs of the Windows build broke:

```
No files were found with the provided path: C:\mixxx-vcpkg/buildtrees/**/*.log. No artifacts will be uploaded.
```

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -12,13 +12,17 @@ jobs:
...
-            #  vcpkg_path: /Users/runner/mixxx-vcpkg
-            #  vcpkg_bootstrap: ./bootstrap-vcpkg.sh
-            #  vcpkg_triplet: x64-osx
-            #  vcpkg_cache: /Users/runner/.cache/vcpkg/archives
-            #  check_disk_space: df -h
+          - os: macos-10.15
+            vcpkg_path: /Users/runner/mixxx-vcpkg
+            vcpkg_bootstrap: ./bootstrap-vcpkg.sh
+            vcpkg_triplet: x64-osx
+            vcpkg_overlay_ports: overlay/x64-osx:overlay/ports
```
Comment (Be-ing) [inline .github/workflows/build.yml:22]: Let's keep the CPU architecture out of the overlay path and use just the OS. Otherwise it will be awkward in the future when we make ARM builds for macOS.

**Comment by Be-ing** [comment]
```
$ cmake -DCMAKE_TOOLCHAIN_FILE=buildenv/mixxx-deps-2.4-x64-osx-96ac0de/scripts/buildsystems/vcpkg.cmake -S . -B build
-- Setting CMAKE_BUILD_TYPE to 'Debug' as none was specified.
-- The C compiler identification is AppleClang 9.0.0.9000039
-- The CXX compiler identification is AppleClang 9.0.0.9000039
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - failed
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang - broken
CMake Error at buildenv/2.3-j00026-0b1209e3-osx10.12-x86_64-release/share/cmake-3.19/Modules/CMakeTestCCompiler.cmake:66 (message):
  The C compiler

    "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /Users/bwilson/mixxx/build/CMakeFiles/CMakeTmp
    
    Run Build Command(s):/usr/bin/make cmTC_6907b/fast && make: error: unable to find utility "make", not a developer tool or in PATH
    
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:61 (project)


-- Configuring incomplete, errors occurred!
```

File: overlay/ports/qt5-base/CONTROL (hunk 1 above)
Comment (daschuer) [reply overlay/ports/qt5-base/CONTROL]: done

File: ports/portmidi/portfile.cmake (hunk 3 above)
Comment (daschuer) [reply ports/portmidi/portfile.cmake]: I don't think so. We need to test.

File: .github/workflows/build.yml (hunk 6 above)
Comment (daschuer) [reply .github/workflows/build.yml:22]: OK

**Comment by daschuer** [comment]
> Somehow uploading the logs of the Windows build broke:

This happens only in case of build errors

**Comment by daschuer** [comment]
```
CMake Error at buildenv/2.3-j00026-0b1209e3-osx10.12-x86_64-release/share/cmake-3.19/Modules/CMakeTestCCompiler.cmake:66 (message):
```
why is this is still 2.3 ?

**Comment by daschuer** [comment]
> Somehow uploading the logs of the Windows build broke

I guess there are non, because building is skipped due to caching.

**Comment by Be-ing** [comment]
> why is this is still 2.3 ?

The vcpkg archive does not include cmake nor ccache, so I am still using those binaries from the old build environment.

**Comment by Be-ing** [comment]
Thank you for rebasing and cleaning up the commit messages. It looks like the new custom triplet is not working unfortunately.

**Comment by Be-ing** [comment]
There is no need to take care of these in this PR, but I would like to ship ccache and cmake with our macOS dependencies so macOS developers only need to download one archive besides the Mixxx source code. I do not know if upstream would accept packages for those because vcpkg is focused on libraries, but it is worth asking.

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -62,7 +62,7 @@ jobs:
         wavpack
         ${{ matrix.vcpkg_packages_extras }}
       VCPKG_DEFAULT_TRIPLET: ${{ matrix.vcpkg_triplet }}
-      VCPKG_OVERLAY_TRIPLETS: overlay/triplets
+      #VCPKG_OVERLAY_TRIPLETS: overlay/triplets does not work
```
Comment (Be-ing) [inline .github/workflows/build.yml:65]: You could open an issue upstream to request this feature.

Comment (daschuer) [reply .github/workflows/build.yml:65]: https://github.com/microsoft/vcpkg/issues/18764

**Comment by daschuer** [comment]
> Run Build Command(s):/usr/bin/make cmTC_6907b/fast && make: error: unable to find utility "make", not a developer tool or in PATH

Is make in your path?

**Comment by daschuer** [comment]
OK, It is building again. Merge? 
This puts us in the situation that we can use the new environment in a PR and sort out remaining issues.

**Comment by Be-ing** [comment]
I would like to get this working locally or in a PR before merging. There may be changes needed in this branch to make it work still.

**Comment by daschuer** [comment]
Yes of cause there are probably remaining issues.
But without merging this current state no one else can help.

Wen we have the environment produced here on our download server, I can also continue to work on it fixing the mixxx repro or this one.

I think this was the way we have developed the Windows version as well.

**Comment by Be-ing** [comment]
If you make a branch on the mixxxdj/vcpkg repo, GitHub Actions will upload the build to downloads.mixxx.org which can be used for testing a PR in the mixxxdj/mixxx repo.

**Comment by daschuer** [comment]
Why do you think we need a branch? We have the same with less hassle when we just merge.

**Comment by Be-ing** [comment]
We can rebase a branch as needed until we merge it.

**Comment by daschuer** [comment]
Please explain the benefit?

**Comment by Be-ing** [comment]
a cleaner git history

There is no rush to merge until we know it works IMO. Do you have permission to push a new branch to this repo?

**Comment by daschuer** [comment]
I don't have interest to do more work for a clean history. This has gone already through a clean up which took me some 
hours. 

What is your current build output? Does installing make fix the issue?

**Comment by daschuer** [comment]
Using an relative path allows to use the original overlays/triplets folder.

**Comment by Holzhaus** [comment]
> a cleaner git history

I think it's okay to merge as-is. This PR builds fine and there is no regression risk because we didn't use vcpkg for macOS until now. If further changes are necessary to make macOS build with vcpkg, the git history may as well reflect that.

**Comment by Be-ing** [comment]
> I don't have interest to do more work for a clean history.

That's fair. I do have interest. I am willing to take over this branch until I get it a macOS build with it working locally.

> This has gone already through a clean up which took me some hours.

Thank you. This will make any more changes to the Git history significantly easier.

**Comment by Be-ing** [comment]
I am generally in favor of having a lower threshold for merging PRs. But not so low that we don't even verify that they work first.

**Comment by daschuer** [comment]
OK, I have pushed it now to https://github.com/mixxxdj/vcpkg/tree/2.3.macos that we can collect changes there. 

Can you please post your build output? What is failing? Did installing make does the trick?

**Comment by Be-ing** [comment]
Why does the upstream ffmpeg package not work?

**Comment by daschuer** [comment]
> Why does the upstream ffmpeg package not work?

I think after a lot testing I have figured out why.
The port  port ships a findFFmpeg.cmake without the imported targets like FFMPEG::<component>

It has missing dependcies to the apple frameworks. I have tried to set the missing propertiess like other ports do to get around the special case code in mixxx. 

It is probably more straight forward to use the Mixxx findffmpeg.cmake directly.

**Comment by Be-ing** [comment]
Is this based on the 2.3 or 2.4 branch? There were some big changes to the FFmpeg package in vcpkg a little bit ago which might not be in 2.3. Anyway, we don't use FFmpeg on macOS currently so let's not stress about it for now.

**Comment by Be-ing** [comment]
Thanks for working on the PortMIDI build. If these hacks still need more work, I think it may be better to fork PortMIDI and write a modern CMakeLists.txt for it from scratch. This would also benefit Tenacity.

**Comment by Be-ing** [comment]
Apparently [FreeBSD has trouble with PortMidi's ancient CMake build system too](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=249348). I think it would be worthwhile to rewrite it.

**Comment by Be-ing** [comment]
You can replace the hacks for PortMIDI's build system with using https://github.com/mixxxdj/portmidi/pull/1

**Comment by Be-ing** [comment]
I have opened a [PR for upstream vcpkg](https://github.com/microsoft/vcpkg/pull/18978) to update PortMidi to our new fork. Until that is merged you can copy it to the overlay.

**Comment by Be-ing** [comment]
Please rebase to remove the hacks for PortMidi's old build system and cherry pick my [updated PortMidi port](https://github.com/Be-ing/vcpkg/commit/7a714b1701a0eaff2b1f352cbf46a66670c1f80c).

**Comment by Be-ing** [comment]
vcpkg now supports [pinning specific old versions of dependencies](https://devblogs.microsoft.com/cppblog/take-control-of-your-vcpkg-dependencies-with-versioning-support/) without needing to backport a specific version of a port to an overlay. Unfortunately that requires using vcpkg with a vcpkg.json manifest which [does not support `vcpkg archive` to export the ZIP archive](https://github.com/microsoft/vcpkg/discussions/17873). Manifest mode can use binary caching but that didn't work well when I tried before with Azure Artifacts because the build environment has to be exactly the same to avoid a rebuild.

File: overlay/ports/fdk-aac/portfile.cmake [hunk 8]
```diff
@@ -1,3 +1,7 @@
+# build the dynamic only so the user can use either the stripped or unstripped 
+# version according to his patent or license rights
+vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY ONLY_DYNAMIC_CRT)
```
Comment (Be-ing) [inline overlay/ports/fdk-aac/portfile.cmake:3]: I don't think this is the right way to handle this and I doubt it will work either. I think we need to build fdk-aac with a different triplet from the rest of the libraries or build everything dynamically. Also I don't think the C runtime linkage is relevant.

**Comment by Be-ing** [comment]
Windows build broke after 4daec35 without a useful error message... :/

**Comment by daschuer** [comment]
> Windows build broke after 4daec35 without a useful error message... :/

This happens due to an upgrade to CMake 3.21.0 on the workflow runner. 
The related bug is here: 
https://github.com/microsoft/vcpkg/issues/18636

This means we can also no longer build our vcpkg 2.3 environment on windows.   :-/

**Comment by Be-ing** [comment]
The bug has been fixed in CMake: https://gitlab.kitware.com/cmake/cmake/-/issues/22444

GitHub should update their Windows VM ASAP. In the meantime maybe we can use https://github.com/marketplace/actions/actions-setup-cmake to upgrade CMake to 3.20.1 or downgrade to 3.19.

**Comment by Be-ing** [comment]
The Windows build is going now... what did you change to fix it?

**Comment by daschuer** [comment]
I have merged from microsoft/vcpkg main to our 2.4 branch and rebased this branch on top of it .

**Comment by Be-ing** [comment]
Ah, thank you. I rebased my PortAudio PR too (#20). I have [asked](https://github.com/actions/virtual-environments/issues/3742#issuecomment-890411409) GitHub Actions to update to CMake 3.21.1 as well.

**Comment by Be-ing** [comment]
Please rebase on the 2.4 branch to include #20.

File: overlay/ports/fdk-aac/portfile.cmake (hunk 8 above)
Comment (Be-ing) [reply overlay/ports/fdk-aac/portfile.cmake:3]: I suppose it does work because I see `fixup_bundle` copied the fdk-aac dylib into the app bundle in https://github.com/daschuer/mixxx/runs/3215138488

However my [PR to add the `he-aac` port feature](https://github.com/microsoft/vcpkg/pull/16007) upstream was just approved and I expect it to be merged soon. So I think we should figure out how to build this dynamically with the upstream port despite the rest of the packages being linked statically. This seems to be possible by hacking a special case into our custom triplet:
https://github.com/microsoft/vcpkg/blob/master/docs/users/triplets.md#per-port-customization

File: overlay/triplets/x64-osx.cmake [hunk 9]
```diff
@@ -0,0 +1,11 @@
+set(VCPKG_TARGET_ARCHITECTURE x64)
+set(VCPKG_CRT_LINKAGE dynamic)
+set(VCPKG_LIBRARY_LINKAGE static)
```
Comment (Be-ing) [inline overlay/triplets/x64-osx.cmake:3]:
```suggestion
set(VCPKG_LIBRARY_LINKAGE static)
# Mixxx loads fdk-aac dynamically at runtime. This allows the user to replace the version of fdk-aac we ship with the patent-encumbered HE-AAC removed with another build which supports HE-AAC.
if(${PORT} MATCHES "fdk-aac")
	set(VCPKG_LIBRARY_LINKAGE dynamic)
else()
```

File: overlay/triplets/x64-osx.cmake [hunk 10]
```diff
@@ -0,0 +1,11 @@
+set(VCPKG_TARGET_ARCHITECTURE x64)
```
Comment (Be-ing) [inline overlay/triplets/x64-osx.cmake:1]: Rename this file to `x64-osx-mixxx` to avoid confusion with the upstream triplet.

**Comment by Be-ing** [comment]
Please make PRs for upstream vcpkg for soundtouch, freetype, and harfbuzz or explain why the changes you made in the overlay wouldn't be appropriate for upstream. I noticed the soundtouch port is outdated as well.

**Comment by Be-ing** [comment]
FWIW the Windows GitHub Actions environment [will be updated to CMake 3.21.1](https://github.com/actions/virtual-environments/issues/3742#issuecomment-891069985) later this week.

File: overlay/triplets/x64-osx-mixxx.cmake [hunk 11]
```diff
@@ -9,3 +17,5 @@ set(VCPKG_OSX_DEPLOYMENT_TARGET 10.12)
 set(VCPKG_C_FLAGS -mmacosx-version-min=10.12)
 set(VCPKG_CXX_FLAGS -mmacosx-version-min=10.12)
 
+
+
```
Comment (Be-ing) [inline overlay/triplets/x64-osx-mixxx.cmake]: stray lines

**Comment by Be-ing** [comment]
My [pull request](https://github.com/microsoft/vcpkg/pull/19270) to fix the jack2 port so PortAudio can find it was merged upstream. Please rebase on microsoft/master and add jack2 to the package list so PortAudio can link it.

**Comment by daschuer** [comment]
@Be-ing: The is an issue with the version database in you upstream PR. I have left a comment. Please have a look.

**Comment by Be-ing** [comment]
I think we can change the builder from macos-10.15 to macos-latest. I used macos-10.15 before because the macos-latest jobs used to take a while to start.

**Comment by Be-ing** [comment]
I cherry-picked my updated portaudio port in the overlay to 2.4. This adds jack2 as a dependency (as well as tre for MinGW + MSVC and pthreads for MSVC) of portaudio so there is no need to manually list jack2 in the package list.

**Comment by Be-ing** [comment]
> The is an issue with the version database in you upstream PR. I have left a comment. Please have a look.

I can't find any issue? It works for me locally and on GitHub Actions. Please try rebasing this PR on the 2.4 branch.

**Comment by daschuer** [comment]
> I think we can change the builder from macos-10.15 to macos-latest. 

Lets go ins small steps. I originally want this PR as close as possible to the original environment to not deal wit too many degrees of freedom. 

Is there a technical need for updating the builder?

**Comment by Be-ing** [comment]
> Is there a technical need for updating the builder?

Not really. I just wanted to ensure that the dependencies can all build on macOS 11. Also GitHub Actions will eventually remove the macOS 10.15 runners so it will need to be switched eventually.

**Comment by Be-ing** [comment]
The harfbuzz port in the overlay is broken:
```
Building package harfbuzz[core]:x64-osx-mixxx...
-- [OVERLAY] Loading triplet configuration from: /Users/runner/mixxx-vcpkg/overlay/triplets/x64-osx-mixxx.cmake
-- Installing port from location: /Users/runner/mixxx-vcpkg/overlay/osx/harfbuzz
CMake Error at overlay/osx/harfbuzz/portfile.cmake:18 (else):
  Flow control statements are not properly nested.
Call Stack (most recent call first):
  scripts/ports.cmake:141 (include)
```

File: overlay/triplets/x64-osx-mixxx.cmake [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+else()
+    set(VCPKG_LIBRARY_LINKAGE static)
+endif()
+
+set(VCPKG_CMAKE_SYSTEM_NAME Darwin)
+set(VCPKG_OSX_ARCHITECTURES x86_64)
+
+set(VCPKG_OSX_DEPLOYMENT_TARGET 10.12)
+set(VCPKG_C_FLAGS -mmacosx-version-min=10.12)
+set(VCPKG_CXX_FLAGS -mmacosx-version-min=10.12)
```
Comment (Be-ing) [inline overlay/triplets/x64-osx-mixxx.cmake:18]:
```suggestion
set(MACOS_MINUMUM 10.12)
set(VCPKG_OSX_DEPLOYMENT_TARGET ${MACOS_MINIMUM})
set(VCPKG_C_FLAGS -mmacosx-version-min=${MACOS_MINIMUM})
set(VCPKG_CXX_FLAGS -mmacosx-version-min=${MACOS_MINIMUM})
```
to ensure consistency

File: .github/workflows/build.yml [hunk 13]
```diff
@@ -30,9 +34,7 @@ jobs:
         hss1394
         libebur128
         libflac
-        libid3tag
         libkeyfinder
-        libmad
```
Comment (Be-ing) [inline .github/workflows/build.yml:35]: Are these not needed for macOS because we use CoreAudio? Ping @uklotzde

Comment (Be-ing) [reply .github/workflows/build.yml:35]: I copied the libid3tag for Tenacity but it does not build for macOS.

File: overlay/triplets/x64-osx-mixxx.cmake (hunk 12 above)
Comment (Be-ing) [reply overlay/triplets/x64-osx-mixxx.cmake:18]: Nevermind, this code does not work. I tried the same for Tenacity and the ffmpeg build failed:
`clang: error: invalid version number in '-mmacosx-version-min='`

File: .github/workflows/build.yml (hunk 13 above)
Comment (daschuer) [reply .github/workflows/build.yml:35]: Yes, it is not used in MacOS.

Comment (Be-ing) [reply .github/workflows/build.yml:35]: okay

**Comment by Be-ing** [comment]
I have rebased to clean up the Git history on my branch: https://github.com/Be-ing/vcpkg/tree/mixxx-macos
If that builds, I think we can merge. Is there anything else we need to take care of first?

**Comment by Be-ing** [comment]
What was the issue with the libusb port? Should that be fixed upstream?

**Comment by daschuer** [comment]
> I have rebased to clean up the Git history on my branch  ...

Why are you doing this? The commits are already quite clean, I am just experimenting which commit is really needed. 
Let's keep the work here.

**Comment by Be-ing** [comment]
There are commits that are canceled out by later commits. Also I rearranged the commits so all of the commits for one port are together.

**Comment by daschuer** [comment]
The libusb overlay fixes this rejected bug: 
https://github.com/microsoft/vcpkg/issues/18442

**Comment by Be-ing** [comment]
What is left to do? Please open a PR for the mixxx repo using this to show it is working.

**Comment by Be-ing** [comment]
Has it been tested by a macOS user?

File: overlay/osx/freetype/CONTROL [hunk 14]
```diff
@@ -2,7 +2,7 @@ Source: freetype
 Version: 2.10.4
 Homepage: https://www.freetype.org/
 Description: A library to render fonts.
-Default-Features: zlib, bzip2, png, brotli
+Default-Features: zlib, bzip2, png
```
Comment (Be-ing) [inline overlay/osx/freetype/CONTROL]: A port should not be in the overlay just to select features; that is the point of features. Instead, explicitly set the features by installing `freetype[zlib, bzip2, png]`.

**Comment by Be-ing** [comment]
My PR to [upstream the fdk-aac fork without HE-AAC](https://github.com/microsoft/vcpkg/pull/16007) was merged. I will take care of removing it from our overlay after merging this.

File: overlay/osx/freetype/CONTROL (hunk 14 above)
Comment (daschuer) [reply overlay/osx/freetype/CONTROL]: I have tried that, but it is ignored.
It is because freetype is include as requiremt with it's default options by other libraries.

Comment (Be-ing) [reply overlay/osx/freetype/CONTROL]: Ah, that makes sense. Which libraries depend on it? If those libraries are in our overlay anyway, we can edit their vcpkg.json manifests to select the features for freetype.

Comment (daschuer) [reply overlay/osx/freetype/CONTROL]: Unfortunately not is is included by many for instance harfbuz

Comment (Be-ing) [reply overlay/osx/freetype/CONTROL]: I'm confused. What is not included?

Comment (daschuer) [reply overlay/osx/freetype/CONTROL]: The patch was merged upstream:
https://github.com/microsoft/vcpkg/pull/19444

Comment (Be-ing) [reply overlay/osx/freetype/CONTROL]: I'm confused what that PR has to do with selecting the features for freetype.

Comment (daschuer) [reply overlay/osx/freetype/CONTROL]: Nothing. That was the reason why harfbuz is no longer in our overlay

Comment (Be-ing) [reply overlay/osx/freetype/CONTROL]: Oh, you mean the libraries that depend on freetype are not in the overlay, right? Is freetype is a dependency of a dependency of Qt? Am I understanding that right?

Comment (daschuer) [reply overlay/osx/freetype/CONTROL]: Yes

Comment (Be-ing) [reply overlay/osx/freetype/CONTROL]: Okay. Please explain in the commit message that we cannot change the port features of the package requiring this because this is a dependency of a dependency of Qt.

Comment (daschuer) [reply overlay/osx/freetype/CONTROL]: Done.

**Comment by Be-ing** [comment]
Looks good to me as soon as we get confirmation from a macOS user that it is working.

**Comment by daschuer** [comment]
We have the confirmation that it works in terms of using the new libraries. I think we can merge and fix the found issues in the mixxx repro, linking to this new environment.

**Comment by Holzhaus** [comment]
Alright, then I'll merge this now so that multiple people can contribute fixes if necessary instead of putting all the weight on @daschuer's shoulders. It won't make a difference for users/developers until we change the buildenv in the mixxx repo anyway.
